workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Text.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using Improbable.Gdk.CodeGeneration.Utils;
+
+namespace Improbable.Gdk.CodeGeneration.CodeWriter
+{
+    /// <summary>
+    /// Free-form text to write.
+    /// </summary>
+    internal class Text : ICodeBlock
+    {
+        private readonly string snippet;
+
+        internal Text(string snippet)
+        {
+            this.snippet = snippet.Trim();
+        }
+
+        public bool HasValue()
+        {
+            return !string.IsNullOrEmpty(snippet);
+        }
+
+        public string Output(int indentLevel)
+        {
+            var spaces = new string(' ', indentLevel * CommonGeneratorUtils.SpacesPerIndent);
+            var indentedCode = snippet
+                .Replace($"{Environment.NewLine}", $"{Environment.NewLine}{spaces}")
+                .Replace($"{Environment.NewLine}{spaces}{Environment.NewLine}", $"{Environment.NewLine}{Environment.NewLine}");
```

Comment:
What's the purpose of the second replace?

Reply:
If your snippet is:
```
line1
line2

line3
```

_imagine underscore represents spaces_

The first replace does this:
```
line1
____line2
____
____line3
```

The second replace does this:
```
line1
____line2

____line3
```

And at the end we return this:
```
____line1
____line2

____line3
```